app/controllers/session_controller.rb

```diff
@@ -424,6 +426,72 @@ def one_time_password
     render layout: 'no_ember', locals: { hide_auth_buttons: true }
   end
 
+  def second_factor_auth_show
+    user = current_user
```

Comment:
You can just use `requires_login` for this, no?

https://github.com/discourse/discourse/blob/b5765ad95b5fa80a09cda4325ed5c3776ddf4440/app/controllers/tags_controller.rb#L14-L21